plugins/outputs/elasticsearch/elasticsearch.go

```diff
@@ -0,0 +1,246 @@
+	"gopkg.in/olivere/elastic.v5"
+)
+
+type Elasticsearch struct {
+	URLs                []string `toml:"urls"`
+	IndexName           string
+	Username            string
+	Password            string
+	EnableSniffer       bool
+	HealthCheckInterval int
+	ManageTemplate      bool
+	TemplateName        string
+	OverwriteTemplate   bool
+	Client              *elastic.Client
+}
+
+var sampleConfig = `
+  ## The full HTTP endpoint URL for your Elasticsearch instance
+  ## Multiple urls can be specified as part of the same cluster,
+  ## this means that only ONE of the urls will be written to each interval.
+  urls = [ "http://node1.es.example.com:9200" ] # required.
+  ## Set to true to ask Elasticsearch a list of all cluster nodes,
+  ## thus it is not necessary to list all nodes in the urls config option
+  enable_sniffer = true
+  ## Set the interval to check if the nodes are available, in seconds
+  ## Setting to 0 will disable the health check (not recommended in production)
+  health_check_interval = 10
+  ## HTTP basic authentication details (eg. when using Shield)
+  # username = "telegraf"
+  # password = "mypassword"
+
+  # Index Config
+  ## The target index for metrics (telegraf will create it if not exists)
+  ## You can use a different index per time frame using the patterns below
+  # %Y - year (2016)
+  # %y - last two digits of year (00..99)
+  # %m - month (01..12)
+  # %d - day of month (e.g., 01)
+  # %H - hour (00..23)
+  index_name = "telegraf-%Y.%m.%d" # required.
+
+  ## Template Config
+  ## set to true if you want telegraf to manage its index template
+  manage_template = true
+  ## The template name used for telegraf indexes
+  template_name = "telegraf"
+  ## set to true if you want to overwrite an existing template
+  overwrite_template = false
+`
+
+func (a *Elasticsearch) Connect() error {
+	if a.URLs == nil || a.IndexName == "" {
+		return fmt.Errorf("Elasticsearch urls or index_name is not defined")
+	}
+
+	ctx := context.Background()
+
+	var clientOptions []elastic.ClientOptionFunc
+
+	clientOptions = append(clientOptions,
+		elastic.SetSniff(a.EnableSniffer),
+		elastic.SetURL(a.URLs...),
+	)
+
+	if a.Username != "" && a.Password != "" {
+		clientOptions = append(clientOptions,
+			elastic.SetBasicAuth(a.Username, a.Password),
+		)
+	}
+
+	if a.HealthCheckInterval > 0 {
+		clientOptions = append(clientOptions,
+			elastic.SetHealthcheckInterval(time.Duration(a.HealthCheckInterval)*time.Second),
+		)
+	}
+
+	client, err := elastic.NewClient(clientOptions...)
+
+	if err != nil {
+		return fmt.Errorf("Elasticsearch connection failed: %s", err)
+	}
+
+	// check for version on first node
+	esVersion, err := client.ElasticsearchVersion(a.URLs[0])
+
+	if err != nil {
+		return fmt.Errorf("Elasticsearch version check failed: %s", err)
+	}
+
+	// warn about ES version
+	if i, err := strconv.Atoi(strings.Split(esVersion, ".")[0]); err == nil {
+		if i < 5 {
+			log.Println("W! Elasticsearch version not supported: " + esVersion)
```

Comment:
Not sure here, I didn't test to really know it is not going to work on older versions. Maybe someone wants to take the risk, so there's the warning. But I agree to return if you think it is safer.